content/upcoming/enry.md

```diff
@@ -0,0 +1,206 @@
+
+```go
+enry.DefaultStrategies = myStrategies
+enry.DefaultClassifier = myClassifier
+```
+
+From this point on, `GetLanguage` and `GetLanguages` will use your custom implementation.
+
+## How does enry know all about a lot of languages?
+
+*enry* uses the information about languages (filenames, interpreters, extensions... associated with each language) that *linguist* keeps on its project.
+
+Specifically, *enry* uses the following files from *linguist*:
+* [languages.yml](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml)
+* [heuristics.rb](https://github.com/github/linguist/blob/master/lib/linguist/heuristics.rb)
+* [vendor.yml](https://github.com/github/linguist/blob/master/lib/linguist/vendor.yml)
+* [documentation.yml](https://github.com/github/linguist/blob/master/lib/linguist/documentation.yml)
+
+These files are parsed to retrieve the necessary information. Then the source files in *enry* that offer this information to the rest of the project are generated and encapsulated as an [internal subpackage data](https://github.com/src-d/enry/tree/master/data).
+
+The whole process is automated and you only need to run `go generate` from the project's root directory to launch it. It allows *enry* to get updated without complex modifications when *linguist* adds new information.
+
+## Does enry actually have better performance?
+
+*enry*'s language detection has been compared with *linguist*.  In order to do that, [linguist's project directory linguist/samples](https://github.com/github/linguist/tree/master/samples) was used as a set of files to run benchmarks against.
+
+The number of language detections for each file in samples directory and per each time interval in a logarithmic scale has been measured for both tools, getting the following results:
+
+```
+enry processed files: 1839
+     1us-10us 5.220228%
+     10us-100us 67.645459%
+     100us-1ms 17.455139%
+     1ms-10ms 7.340946%
+     10ms-100ms 2.338227%
+
+linguist processed files: 1839
+     1us-10us 0.000000%
+     10us-100us 4.023926%
+     100us-1ms 50.027189%
+     1ms-10ms 42.849375%
+     10ms-100ms 3.099511%
+```
+
+As you can see, *enry* was able to detect 72% of files in a time between 1us and 100us, while the majority of the files *linguist* processed are shifted to greater time intervals.
+
+Calculating the mean spent time to process a file for both tools, on average *enry* is 211% faster than *linguist*.
+
+Considering that *enry* follows the same algorithms that *linguist* does, it looks like the performance improvement is given by the chosen language to develop them.
+
+However, it should be noted that in a few cases *enry* could turn slower than linguist. This is due to Golang's regexp being slower than Ruby's, which uses [oniguruma](https://github.com/kkos/oniguruma) library, written in C.
```

Comment:
I am strongly convinced that you need to run `go profile` and list the profile here. Most of the time is likely to be spent on IO or regexp, but at least there will be the proof.

It would be super awesome hyper cool to include the analogous profile for Ruby.